pkg/trace/call_test.go

```diff
@@ -39,142 +39,113 @@ func (suite) TestNestedCall(t *testing.T) {
-	func() {
-		ctx = trace.StartTrace(ctx, "trace-test")
-		defer trace.End(ctx)
-
-		if err := outer(ctx, &Model{ID: "some model id"}); err == nil {
-			t.Fatal("unexpected success", err)
-		}
-	}()
-
 	// don't care about specific ids but make sure same IDs are used in both settings
 	traceID, rootID, middleID := differs.CaptureString(), differs.CaptureString(), differs.CaptureString()
 
 	expected := []map[string]interface{}{
 		{
-			"app.name":             "gobox",
-			"app.version":          "testing",
-			"sql.query":            "my query: some model id",
-			"model.id":             "some model id",
-			"duration_ms":          differs.FloatRange(0, 2),
-			"meta.beeline_version": differs.AnyString(),
-			"meta.local_hostname":  differs.AnyString(),
-			"meta.span_type":       "leaf",
-			"name":                 "sql",
-			"service_name":         "log-testing",
-			"error.kind":           "error",
-			"error.error":          "sql error",
-			"error.message":        "sql error",
-			"error.stack":          differs.AnyString(),
-			"trace.parent_id":      middleID,
-			"trace.span_id":        differs.AnyString(),
-			"trace.trace_id":       traceID,
-			"info_1_key":           "info_1_val",
-			"info_2_key":           "info_2_val",
-			"timing.dequeued_at":   differs.RFC3339NanoTime(),
-			"timing.finished_at":   differs.RFC3339NanoTime(),
-			"timing.scheduled_at":  differs.RFC3339NanoTime(),
-			"timing.service_time":  differs.FloatRange(0, 1),
-			"timing.total_time":    differs.FloatRange(0, 1),
-			"timing.wait_time":     differs.FloatRange(0, 1),
+			"name":                           "sql",
+			"spanContext.traceID":            traceID,
+			"spanContext.spanID":             differs.AnyString(),
+			"spanContext.traceFlags":         "01",
+			"parent.traceID":                 traceID,
+			"parent.spanID":                  middleID,
+			"parent.traceFlags":              "01",
+			"parent.remote":                  false,
+			"spanKind":                       "internal",
+			"startTime":                      differs.AnyString(),
+			"endTime":                        differs.AnyString(),
+			"attributes.app.name":            "gobox",
+			"attributes.service_name":        "gobox",
+			"attributes.app.version":         "testing",
+			"attributes.sql.query":           "my query: some model id",
+			"attributes.model.id":            "some model id",
+			"attributes.error.kind":          "error",
+			"attributes.error.error":         "sql error",
+			"attributes.error.message":       "sql error",
+			"attributes.error.stack":         differs.AnyString(),
+			"attributes.info_1_key":          "info_1_val",
+			"attributes.info_2_key":          "info_2_val",
+			"attributes.timing.dequeued_at":  differs.RFC3339NanoTime(),
+			"attributes.timing.finished_at":  differs.RFC3339NanoTime(),
+			"attributes.timing.scheduled_at": differs.RFC3339NanoTime(),
+			"attributes.timing.service_time": differs.AnyString(),
```

Comment:
Yeah, this is something Richard mentioned too. So OTel attributes require you to specify the type. Honeycomb attributes took an `interface{}` and handled that correctly. The way we add attributes using a logger passes an `interface{}` everywhere. 

How the logger works left me with two options: cast everything to the same type, I chose string since I can just use `fmt.Sprinf`, or do some reflection magic to determine the type. As this PR was already getting complex, I went for the simpler solution. But I am happy to update it in a follow up to use reflection and set the right types.

From the Honeycomb perspective, it looks like Honeycomb was smart to automatically convert this string.